avocado/core/loader.py

```diff
@@ -44,6 +44,52 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags, ignore_empty=False):
+    """
+    Filter the existing (unfiltered) test suite based on tags
+
+    The filtering mechanism is limited to INSTRUMENTED tests, that is it does
+    not apply to SIMPLE tests or any other test type.  It mean non-INSTRUMENTED
+    tests are *always* included.
```

Comment:
This is actually not true. It's a generic concept and it works for all tests where the loader can fill in the tags attribute. It could be implemented even for avocado-vt and we could introduce tags for SIMPLE tests as well (not likely, but could be). I'd just say that it filters test-suite and only includes tests which are tagged by loader accordingly to the filters. For tests which does not have tags specified the behavior depends on `ignore_empty`.